### ISSUE TITLE
Convert view to model using positions

### ISSUE THREAD

**Issue by pjasiun** [ckeditor/ckeditor5-engine#1213]
Context: view to model conversion.

Right now converters build nodes which are passed to the parent converters which build parent element with children it gets as an array of elements. There are couple problems with this:
 - the structure is not built until all top-level elements are converted, we can not use live ranges during the conversion because of it,
 - we need to pass "context" object to children which know nothing more about the structure than this context, can not check siblings, also context needs to be updated in every converter.

After closing https://github.com/ckeditor/ckeditor5-engine/issues/858 we should try to fix it.

Parent converter should pass to the child converter position which is the conversion context. Child converter should return the range with the converted fragment. Parent converter can execute its action before or after child converters.

Consider an example: `<p><b>foo</b></p>`

- `<p>` converter creates `<paragraph>` and pass the position inside it to `<b>` converter (`<paragraph>^</paragraph>`),
- `<b>` converter call `foo` converter first and pass the position it gets from `<p>` converter,
- `foo` converter insert text on the position and returns range with inserted element (`<paragraph>[foo]</paragraph>`),
- `<b>` will add attribute to the range.

**Comment by Reinmar** [comment]
This has a huge 👍  from me.

And this is an assumption which I made when implementing the new Schema. Its `checkAttribute()` method accepts a real node as its context. It doesn't accept anymore a fake `context` arrays of element names in addition to that node. This fake `context` arrays were used so far in `buildviewconverter.js`'s `setAttributeOn()` to check whether this attribute can be applied in this specific context.

Since I removed the support for fake contexts (which are bad because they are fake so they don't carry all the information) I now need to fully depend on the passed node. Unfortunately, right now the conversion first try to apply the attribute (on a `DF>$text`) and only then insert it to the tree. Which means that `checkAttribute( node, attrName )` will not know the whole context.

Once the conversion will first insert the text node to the model and then trigger attribute application, this will work perfectly fine with the new Schema.

But for now, I need to comment this test out because I can't make it pass easily:

https://github.com/ckeditor/ckeditor5-engine/blob/aea6119b12b5e3d009ff8597c3546ab8b488fe88/tests/conversion/buildviewconverter.js#L499-L516

PS. This is only one reason why I like the idea to use positions. The other one is – it's just so much easier to think this way ;)
